docs/site/Parsing-requests.md

```diff
@@ -0,0 +1,202 @@
+
+{% include note.html title="OpenAPI operation object" content="See
+[OpenAPI operation object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#operationObject)
+to know more about its structure." %}
+
+### Coercion
+
+The parameters parsed from path, header, and query of a http request are always
+in the string format when using the `http` module in Node.js to handle requests.
+Therefore when invoking a controller function, a parameter need to be converted
+to its corresponding JavaScript runtime type, which is inferred from its
+parameter specification.
+
+Take the example operation `replaceTodo` in section
+[parsing raw data](#parsing-raw-data) which takes in a number `id` as the first
+input. Without coercion,`id` would have to be manually cast into the number type
+before it can be used as seen below:
+
+```ts
+@put('/todos/{id}')
+async replaceTodo(
+  @param.path.number('id') id: number,
+  @requestBody() todo: Todo,
+): Promise<boolean> {
+  // NO need to do the "string to number" convertion now,
+  // coercion automatically handles it for you.
+  id = +id;
+  return await this.todoRepo.replaceById(id, todo);
+}
+```
+
+### Validation
+
+Validations are applied on the parameters and the request body data. They also
+use OpenAPI specification as the reference to infer the validation rules.
+
+#### Parameters
+
+We have the data type safety check for the parameters parsed from header, path,
+and query. For example, if a parameter should be an integer, then a number with
+decimal like "1.23" would be rejected.
+
+You can specify a parameter's type by calling shortcut decorators of `@param`
+like `@param.query.integer()`. A list of available shortcuts can be found in the
+[api documents](https://apidocs.strongloop.com/@loopback%2fdocs/openapi-v3.html#param).
+Check out the section on
+[parameter decorators](Decorators.md#parameter-decorator) for instructions on
+how to decorate the controller parameter.
+
+Here are our default validation rules for each type:
+
+- number: validated by `isNaN(Number(data))`.
+- integer: validated by `Number.isInteger(data)`.
+- long: validated by `Number.isSafeInteger(data)`.
+- date-time: should be a valid date-time defined in
+  [RFC3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14).
+- date: should be a valid full-date defined in
+  [RFC3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14).
+- boolean: after converted to all upper case, should be one of the following
+  values: `TRUE`, `1`, `FALSE` or `0`.
+
+#### Request Body
+
+The data from request body is validated against its OpenAPI schema
+specification. We use module [AJV](https://github.com/epoberezkin/ajv) to
+perform the validation, which validates data with a JSON schema generated from
+the OpenAPI schema specification.
+
+Take again the operation replaceTodo for instance:
+
+```ts
+import {Todo} from './models';
+
+...
```

Comment:
minor nitpick: can we wrap this code block with the controller class declaration?

```ts
import {Todo} from './models';

class test {
  ...
  @put('/todos/{id}')
  async replaceTodo(
    @param.path.number('id') id: number,
    @requestBody() todo: Todo,
  ): Promise<boolean> {
    return await this.todoRepo.replaceById(id, todo);
  }
  ...
}

```

it just didn't render properly for me in VS code :P

Reply:
There are too many irrelevant code in the class definition which distract users when reading :( that's why I deleted them.

> it just didn't render properly for me in VS code

Mine looks good 🤔 :
<img width="485" alt="screen shot 2018-07-13 at 3 11 18 pm" src="https://user-images.githubusercontent.com/12554153/42709585-04db9420-86af-11e8-9d8f-b6c876adac3d.png">